tests/test_bulk_upload_documents.py

```diff
@@ -0,0 +1,71 @@
+import os
+import tempfile
+import shutil
+
+from sys import version_info
+if version_info.major == 2:
+    import __builtin__ as builtins
+else:
+    import builtins
+
+from dmscripts.bulk_upload_documents import get_bucket_name, get_all_files_of_type, \
+    get_supplier_id_from_framework_file_path, upload_file
+import mock
+
+
+def test_get_bucket_name_for_agreements_documents():
+    assert get_bucket_name('dev', 'agreements') == 'digitalmarketplace-agreements-dev-dev'  # noqa
+    assert get_bucket_name('preview', 'agreements') == 'digitalmarketplace-agreements-preview-preview'  # noqa
+    assert get_bucket_name('staging', 'agreements') == 'digitalmarketplace-agreements-staging-staging'  # noqa
+
+
+def test_get_all_files_of_type_for_flat_folder():
+    temp_folder_path = tempfile.mkdtemp()
+    pdf1 = open(os.path.join(temp_folder_path, 'test1.pdf'), 'w+')
+    pdf2 = open(os.path.join(temp_folder_path, 'test2.pdf'), 'w+')
+    assert len(list(get_all_files_of_type(temp_folder_path, 'pdf'))) == 2
+    pdf1.close()
+    pdf2.close()
+    shutil.rmtree(temp_folder_path)
+
+
+def test_get_all_files_of_type_for_nested_folder():
+    temp_folder_path = tempfile.mkdtemp()
+    tempfile.tempdir = temp_folder_path
+    pdf1 = open(os.path.join(temp_folder_path, 'test1.pdf'), 'w+')
+    nested_temp_folder_path = tempfile.mkdtemp()
+    pdf2 = open(os.path.join(nested_temp_folder_path, 'test2.pdf'), 'w+')
+    assert len(list(get_all_files_of_type(temp_folder_path, 'pdf'))) == 2
+    pdf1.close()
+    pdf2.close()
+    shutil.rmtree(temp_folder_path)
+
+
+def test_get_supplier_id_from_framework_file_path_for_file_with_dashes():
+    path = '../../Downloads/Completed Frameworks BATCH2 SHARED WITH GDS/ICNT_Consulting_Ltd-584425-signed-framework-agreement.pdf'  # noqa
+    assert get_supplier_id_from_framework_file_path(path) == '584425'
+
+
+def test_get_supplier_id_from_framework_file_path_for_file_with_underscores():
+    path = '../../Downloads/Completed Frameworks BATCH2 SHARED WITH GDS/ICNT_Consulting_Ltd-584425_signed_framework_agreement.pdf'  # noqa
+    assert get_supplier_id_from_framework_file_path(path) == '584425'
+
+
+@mock.patch('dmutils.s3.S3')
+def test_upload_file(bucket):
+    file_path = 'The_Business_Software_Centre-92877-signed-framework-agreement.pdf'
```

Comment:
Uploaded file names should no longer include supplier names, so I think it's worth updating the tests to reflect the correct format: e.g. `<supplier_id>-signed-framework-agreement.pdf`